### DISCUSSION TITLE
Access PrismaClient in NextAuth in frontend

### DISCUSSION THREAD

**Discussion by fpatzig** [prisma/prisma#17037] [Q&A]
Hi all,

I am using NextAuth in my frontend and I want to import prisma from my Nest backend to to use the PrismaAdapter, i.e.,

`[...nextauth].js`
`import prisma from '../../../../server/lib/prisma.ts';`
`...`
`adapter: PrismaAdapter(prisma)`

In my server folder I created a lib folder in which I have a `prisma.ts` file. In this file I set up a global prisma variable, i.e.,

`import { PrismaClient } from "@prisma/client"`

`declare global {`
`  var prisma: PrismaClient | undefined`
`}`

`const client = global.prisma || new PrismaClient()`
`if (process.env.NODE_ENV !== "production") global.prisma = client`

`export default client`

If I now run the code and try to access this prisma variable I receive a module parse error:

ModuleParseError: Module parse failed: Unexpected token (3:8)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See [https://webpack.js.org/concepts#loaders](https://webpack.js.org/concepts#loaders|)

Does anyone has an idea on how to solve this problem and get access to the global prisma variable in my frontend so that I can use NextAuth there?

Any help is appreciated!

Thank you.

**Comment by pantharshit00** [comment]
Hello @fpatzig,
  
Can you please share your setup in a small git repository? 

Also, you can checkout this example that we have which uses nextauth: https://github.com/prisma/prisma-examples/tree/latest/typescript/rest-nextjs-api-routes-auth

**Reply by fpatzig** [reply]
Hi @pantharshit00,

Thanks a lot for your reply!

I created a new project where I replicate the problem / error. It can be found here: https://github.com/fpatzig/next-nest-fullstack-prisma-problem-demo.git

**I give a detailed description of what I did and how I came to the problem:**

1. Initialize new next.js project called 'client': npx create-next-app

2. Initialize new nest.js project called 'server': npx nest new server

3. Set up docker-compose.yml file on server side
- User: postgres
- Password: postgres
- Database: test-db
- Port: 5432:5434

4. Launch the postgresql database in docker on the server:
`server % docker compose up test-db -d`

5. Adapt the .env file on the server side and add the Database URL for a postgres db:
`DATABASE_URL=„postgresql://postgres:postgres@localhost:5432/nest?schema=public"`

6. Install prisma and prisma CLI on the server side; initialize prisma on the server side:
```
server % npm i prisma
server % npm i @prisma/client
server % npx prisma init
```
7. Set up prisma schema, run prisma migration and generate prisma client on server side: 
```
server % npx prisma migrate dev
server % npx prisma generate
```

8. Initialize prisma module on server side:
```
server % nest g module prisma
server % nest g service prisma
```
and delete the prisma.service.spec.ts as it is not needed for now.

9. Set up prisma.module and prisma.service (see repository from above)

10. Creating a lib folder on the server side as in the Nextauth project that you were referring to.

_Now I am moving to the frontend, the next.js project. I would like to access the PrismaClient from the frontend as it is required for the PrisamAdapter in nextauth._

11. So, I set up a [...nextauth].ts file in client/pages/api/auth and fill it as in the project that you were referring to. Now, I adjust the path for the prisma import to:
`import prisma from '../../../../server/lib/prisma'`

`If I now go to http://localhost:3000/api/auth/signin the following error is thrown:`

<img width="964" alt="Bildschirm­foto 2023-01-07 um 12 12 59" src="https://user-images.githubusercontent.com/60009843/211148981-df85f9f9-d646-43ba-a17f-9882137c6b54.png">

My question would be in general: How do I correctly access the PrismaClient that lives on the server side on the client side of my application?

Thanks a lot for your help in advane!

Best Regards
Franz

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
